clientlibs/java/src/main/java/org/upgradeplatform/client/ExperimentClient.java

```diff
@@ -256,48 +308,74 @@ private ExperimentsResponse findExperimentResponse(String site, String target,
 				.orElse(new ExperimentsResponse());
 	}
 
-	private static ExperimentsResponse copyExperimentResponse(ExperimentsResponse experimentsResponse) {
-		Condition assignedCondition = new Condition(experimentsResponse.getAssignedCondition().getId(),
-				experimentsResponse.getAssignedCondition().getConditionCode(), experimentsResponse.getAssignedCondition().getExperimentId(),
-				experimentsResponse.getAssignedCondition().getPayload());
+	private void rotateConditions(String site, String target) {
+		if (this.allExperiments != null) {
+			ExperimentsResponse result = this.allExperiments.stream().filter(t -> t.getSite().equalsIgnoreCase(site) &&
+					(isStringNull(target) ? isStringNull(t.getTarget().toString())
+					: t.getTarget().toString().equalsIgnoreCase(target))).findFirst().orElse(null);
+			if (result != null) {
+				Condition[] rotatedCondition = Arrays.copyOf(result.getAssignedCondition(),
+						result.getAssignedCondition().length);
+				List<Condition> rotatedList = Arrays.asList(rotatedCondition);
+				Collections.rotate(rotatedList, -1);
+				rotatedCondition = rotatedList.toArray(rotatedCondition);
+				result.setAssignedCondition(rotatedCondition);
+				result.setAssignedFactor(result.getAssignedFactor());
```

Comment:
don't have time for a full review, and I mentioned this on another PR for the JS client, but: this kind of business logic in the client library feels like a code smell to me.  Why can't this condition rotation happen on the server?